snaps/how-to/integrate-custom-evm-accounts.md

```diff
@@ -0,0 +1,206 @@
+
+The MetaMask account selection modal has an option called "Add snap account":
+
+![add snap account option](../assets/keyring/add-snap-account.png)
+
+This option will show a list of keyring snaps, but will ultimately redirect the user to the companion Dapp for your Keyring snap. That is, a Dapp that contains all the UI to configure and manage the Keyring snap.
+
+On that Dapp, you'll present a custom user interface allowing the user configure their custom EVM account. The Dapp will make use of the [`KeyringSnapRpcClient`'s `createAccount` method](../reference/keyring-api/classes/KeyringSnapRpcClient.md#createaccount), which will call your `Keyring` class' method of the same name. An example of this can be found in the [Simple keyring snap companion Dapp](https://github.com/MetaMask/snap-simple-keyring/blob/d3f7f0156c59059c995fea87f90a3d0ad3a4c135/packages/site/src/pages/index.tsx#L136).
+
+In your `Keyring` class' `createAccount` method, your responsibility is to create the account based on the parameters that were passed to you. Your snap has to keep track of the accounts that it creates, which can be done using [`snap_manageState`](../reference/rpc-api.md#snap_managestate). Once your snap has created an account, it should notify MetaMask using the [`snap_manageAccounts` `createAccount` method](../reference/rpc-api.md#createaccount). An example of this process can be found in the [Simple keyring snap code](https://github.com/MetaMask/snap-simple-keyring/blob/d3f7f0156c59059c995fea87f90a3d0ad3a4c135/packages/snap/src/keyring.ts#L61).
+
+Once your snap has created an account, that account can be used to sign messages and transactions. In the following sections we'll look at how this can be done.
+
+## Synchronous signing flow
+
+Your Keyring snap will implement the simple flow if it's able to sign transactions directly. This would be the case if the snap doesn't need a third-party such as a hardware key or a second account's signature, as would be the case for a threshold signature scheme. The flow would look like this:
+
+![Synchronous signing flow](../assets/keyring/synchronous-flow.png)
+
+For a full example of a simple keyring snap, see [`snap-simple-keyring` on GitHub](https://github.com/MetaMask/snap-simple-keyring).
+
+
+The flow starts when a Dapp calls a method such as `personal_sign` or `eth_signTransaction`, or when the user initiates a new funds transfer from the MetaMask UI. At that point, MetaMask will detect that this interaction is requested for an account controlled by your Keyring snap.
+
+After the user approves the transaction in the UI, MetaMask will call the `submitRequest` method of your `Keyring` class. `submitRequest` will receive the original RPC request, and will need to return a [`SubmitRequestResponse`](../reference/keyring-api/modules.md#submitrequestresponsestruct) with `pending` set to `false`, and `result` set to the requested signature.
```

Comment:
Also a 404 @ https://github.com/MetaMask/metamask-docs/blob/zs/keyring-doc/snaps/reference/keyring-api/modules.md#submitrequestresponsestruct

Reply:
That's normal, as it gets created at build time from a Git submodule!